koltinx-serialization-scale/src/main/java/io/novasama/substrate_sdk_android/koltinx_serialization_scale/decoder/StructDecoder.kt

```diff
@@ -25,10 +26,14 @@ class StructDecoder(
         // TODO this is a temp solution to cater both camel and snake cases in struct fields
         // This should be addressed better to introducing some annotation to specify field naming strategy per entire struct
         // Also we should check the underlying layer to find out why some fields are not transformed to camel case
+        val camelCaseKey = key.snakeCaseToCamelCase()
         val snakeCaseKey = key.camelCaseToSnakeCase()
-        if (snakeCaseKey in value) return value[snakeCaseKey]
 
-        error("Key '$key' (or '$snakeCaseKey') is not found in the $value")
+        return when {
+            camelCaseKey in value -> value[camelCaseKey]
+            snakeCaseKey in value -> value[snakeCaseKey]
+            else -> error("Key '$key' ('$camelCaseKey'/'$snakeCaseKey') is not found in the $value")
+        }
```

Comment:
I would only transform key when we actually need to check it to avoid extra compute and allocations since this is a hot path. So we first compute `snakeCaseToCamelCase`, check it, and only then compute&check `camelCaseToSnakeCase`. This will avoid extra compute and allocations esp given that this is a hot path